### PR TITLE
feat(portex): add "Portex.from_pyarrow" classmethod

### PR DESCRIPTION
**Pull request by Lee-000**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-296bvcy 实现 PortexType.from_pyarrow 方法](https://app.clickup.com/t/296bvcy)

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/portex/base.py [hunk 1]
```diff
@@ -18,6 +18,20 @@
 
 _INDENT = " " * 2
 
+_PYARROW_TYPE_ID_TO_PORTEX_TYPE_NAME = {
```
[the remark this answers is not in the document]
Comment (Lee-000) [reply graviti/portex/base.py:21]: Get the portex type from `PortexType.imports`? Would it be confusing?
Also, there are still `if` statements in `from_pyarrow` to tell if the type is a `record` or `array`. If portex types are used, `isinstance` will be used and the type will be retrieved again.

File: graviti/portex/base.py [hunk 2]
```diff
@@ -18,6 +29,36 @@
 
 _INDENT = " " * 2
 
+_T = TypeVar("_T", bound="PortexType")
+
+PYARROW_TYPE_ID_TO_PORTEX_TYPE = {}
+
+
+class RegisterPortexType:
```
Comment (linjiX) [inline graviti/portex/base.py:37]: will it be better if making `PYARROW_TYPE_ID_TO_PORTEX_TYPE ` and `_get_pyarrow_type_id` as the member of `RegisterPortexType `

Comment (linjiX) [reply graviti/portex/base.py:37]: Also the name `RegisterPortexType ` seems unrelated with pyarrow?

File: graviti/portex/base.py [hunk 3]
```diff
@@ -98,7 +152,7 @@ def from_pyobj(
         if _imports is None:
             _imports = Imports.from_pyobj(content.get("imports", []))
 
-        class_ = _imports[content["type"]]
+        class_: Type[_T] = _imports[content["type"]]  # type: ignore[assignment]
```
Comment (linjiX) [inline graviti/portex/base.py:155]: where this ignore comes from?

File: graviti/portex/builtin.py [hunk 4]
```diff
@@ -403,6 +431,15 @@ def __init__(
     ) -> None:
         super().__init__(items=items, length=length, nullable=nullable)
 
+    @classmethod
+    def _from_pyarrow(cls, pyarrow_type: pa.DataType) -> "array":
+        value_type = pyarrow_type.value_type
+        value_type_id = cls._get_pyarrow_type_id(value_type)
```
Comment (linjiX) [inline graviti/portex/builtin.py:437]: why not get the type directly from the method rather than get id than get type?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/portex/base.py [hunk 5]
```diff
@@ -18,6 +29,36 @@
...
+
+class RegisterPortexType:
+    """Register the Portex type to set the conversion from PyArrow to Portex.
+
+    Arguments:
+        pyarrow_type_ids: The id of the corresponding PyArrow types.
+
+    """
+
+    def __init__(self, pyarrow_type_ids: Iterable[int]) -> None:
```
Comment (linjiX) [inline graviti/portex/base.py:45]: will `*pyarrow_type_ids` be better?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/portex/base.py (hunk 3 above)
Comment (Lee-000) [reply graviti/portex/base.py:155]: mypy considers that the expression has type "Type[PortexType]".

File: graviti/portex/builtin.py [hunk 6]
```diff
@@ -297,6 +311,10 @@ def __init__(
     ) -> None:
         super().__init__(fields=fields, nullable=nullable)
 
+    @classmethod
+    def _from_pyarrow(cls, pyarrow_type: pa.DataType) -> "record":
```
Comment (linjiX) [inline graviti/portex/builtin.py:315]: should it be `pa.StructType`?

File: graviti/portex/builtin.py [hunk 7]
```diff
@@ -403,6 +424,13 @@ def __init__(
     ) -> None:
         super().__init__(items=items, length=length, nullable=nullable)
 
+    @classmethod
+    def _from_pyarrow(cls, pyarrow_type: pa.DataType) -> "array":
```
Comment (linjiX) [inline graviti/portex/builtin.py:428]: `pa.ListType`?

File: graviti/portex/base.py [hunk 8]
```diff
@@ -110,6 +144,33 @@ def from_pyobj(
...
+        pyarrow_type_id = (
+            pyarrow_type.value_type.id
+            if pyarrow_type.id == pa.lib.Type_DICTIONARY  # pylint: disable=c-extension-no-member
+            else pyarrow_type.id
+        )
+
+        try:
+            portex_type = PYARROW_TYPE_ID_TO_PORTEX_TYPE[pyarrow_type_id]
+        except KeyError:
+            raise TypeError("Not supported PyArrow type") from None
```
Comment (linjiX) [inline graviti/portex/base.py:170]: shall we print the pyarrow type here?

File: graviti/portex/base.py [hunk 9]
```diff
@@ -18,6 +18,36 @@
 
 _INDENT = " " * 2
 
+_T = TypeVar("_T", bound="PortexType")
+
+PYARROW_TYPE_ID_TO_PORTEX_TYPE = {}
+
+
+class RegisterPyArrowConversion:
```
Comment (linjiX) [inline graviti/portex/base.py:26]: will `PyArrowConversionRegister` be better?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/portex/builtin.py [hunk 10]
```diff
@@ -403,6 +424,13 @@ def __init__(
     ) -> None:
         super().__init__(items=items, length=length, nullable=nullable)
 
+    @classmethod
+    def _from_pyarrow(cls, pyarrow_type: Union[pa.ListType, pa.FixedSizeListType]) -> "array":
```
Comment (linjiX) [inline graviti/portex/builtin.py:428]: shall we use TypeVar here?

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**